Firmware/ultralcd.cpp

```diff
@@ -8959,6 +8967,43 @@ void lcd_experimental_menu()
     MENU_END();
 }
 
+void reprint_from_eeprom() {
+	char cmd[30];
+	char filename[13];
+	uint8_t depth = 0;
+	char dir_name[9];
+
+	isPrintFinished=false;
+
+	//cmdqueue_reset();
+
+	depth = eeprom_read_byte((uint8_t*)EEPROM_DIR_DEPTH);
+	
+	MYSERIAL.println(int(depth));
+	for (int i = 0; i < depth; i++) {
+		for (int j = 0; j < 8; j++) {
```

Comment:
how can you be sure that each dir has 8 characters and then there is '/'?

Reply:
Because the EEPROM_DIRS has 80 chars and EEPROM_DIR_DEPTH has a limitation of 10 levels so each _dir name_ can be only 8 chars, I didn't read how the code manage when a dirname is smaller than 8 chars, I suppose that will fill the empty positions with "\0" but I didn't dig on it, as far as I know, the card.dirname[][] store the dirnames associated with the depth of each dir so the function **menu_action_sdfile()**  store in EEPROM_DIRS all the names together and in order to compose the path you need to do the same taking blocks of 8 chars for the dirname. 

```
static void menu_action_sdfile(const char* filename)
{
  loading_flag = false;
  char cmd[30];
  char* c;
  bool result = true;
  sprintf_P(cmd, PSTR("M23 %s"), filename);
  for (c = &cmd[4]; *c; c++)
    *c = tolower(*c);

  const char end[5] = ".gco";

  //we are storing just first 8 characters of 8.3 filename assuming that extension is always ".gco"
  for (uint_least8_t i = 0; i < 8; i++) {
	  if (strcmp((cmd + i + 4), end) == 0) {
		  //filename is shorter then 8.3, store '\0' character on position where ".gco" string was found to terminate stored string properly
 		  eeprom_write_byte((uint8_t*)EEPROM_FILENAME + i, '\0');
		  break;
	  }
	  else {
		  eeprom_write_byte((uint8_t*)EEPROM_FILENAME + i, cmd[i + 4]);
	  }
  }

  uint8_t depth = (uint8_t)card.getWorkDirDepth();
  eeprom_write_byte((uint8_t*)EEPROM_DIR_DEPTH, depth);

  for (uint_least8_t i = 0; i < depth; i++) {
	  for (uint_least8_t j = 0; j < 8; j++) {
		  eeprom_write_byte((uint8_t*)EEPROM_DIRS + j + 8 * i, card.dir_names[i][j]);
	  }
  }
```

Also is the same logic is used on **restore_print_from_eeprom()** so if there is a problem to retrieve the path from the reprint function maybe will have the same behaviour on the recovery from a power outage.

Reply:
ok, you are right